pkg/server/restart.go

```diff
@@ -157,118 +172,128 @@ func (c *criService) loadContainer(ctx context.Context, cntr containerd.Containe
 	}
 	meta := data.(*containerstore.Metadata)
 
+	var containerIO *cio.ContainerIO
 	// Load status from checkpoint.
 	status, err := containerstore.LoadStatus(containerDir, id)
 	if err != nil {
 		logrus.WithError(err).Warnf("Failed to load container status for %q", id)
 		status = unknownContainerStatus()
 	}
 
-	// Load up-to-date status from containerd.
-	var containerIO *cio.ContainerIO
-	t, err := cntr.Task(ctx, func(fifos *containerdio.FIFOSet) (_ containerdio.IO, err error) {
-		stdoutWC, stderrWC, err := c.createContainerLoggers(meta.LogPath, meta.Config.GetTty())
-		if err != nil {
-			return nil, err
-		}
-		defer func() {
+	// Generate container status based on checkedpointed status and actual state in containerd.
```

Comment:
I didn't change any logic, wrap the status generation logic in a function.

Reply:
why the edit?